### DISCUSSION TITLE
Teensy 4 controller area network (CAN) bus support

### DISCUSSION THREAD

**Discussion by jmpolom** [zephyrproject-rtos/zephyr#69052] [General] [answered]
I see that the Teensy 4 is listed as a [compatible board](https://docs.zephyrproject.org/latest/boards/arm/teensy4/doc/index.html) in the docs. The NXP RT1060 SOC on the Teensy 4 board has two CAN adapters however they are not listed as supported in the feature matrix. The NXP RT1060 evaluation kit board uses basically the same SOC and is listed as [supported](https://docs.zephyrproject.org/latest/boards/arm/mimxrt1060_evk/doc/index.html) with the CAN adapter also supported.

Am I wrong in thinking it should be relatively straightforward to enable CAN on the Teensy 4 given that CAN is supported on a very similar board?

**Comment by henrikbrixandersen** [comment, marked as answer]
Yes, that should be pretty straight-forward. I am using the `mimxrt1060_evkb` with CAN here, so it definitely works.
Have a look at https://github.com/zephyrproject-rtos/zephyr/blob/main/boards/arm/mimxrt1060_evk/mimxrt1060_evk.dts#L275 for inspiration on how to enable it for the Teensy 4.

**Reply by jmpolom** [reply]
Pretty much what I thought it might be like. Thank you for the confirmation.
